discovery_engine/lib/src/domain/ai_state_holder.dart

```diff
@@ -0,0 +1,18 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+/// Interface implemented by anything holding AI state.
+abstract class AIStateHolder {
+  Future<void> clearAIState();
+}
```

Comment:
For now this was the simplest way to abstract over multiple places which need to be cleared.

But I noticed that `FeedManager`, `EventHandler` and the other `Managers`  are not really distinct objects in the classical sense as all of their state is the same shared overlapping state (the repos, engine, etc.). They are more different categories of implementation based on the same state object.

This makes me wonder if it makes sense to not make them separate classes but extensions or better mixins into the same `System` class (or more specific the class we currently call EventHandler and then could rename to mixin).

But  this is out of scope of this PR.